pkg/util/encode/encode.go

```diff
@@ -63,6 +65,6 @@ func EncoderFor(format string) (runtime.Encoder, error) {
 	} else {
 		encoder = serializerInfo.Serializer
 	}
-	encoder = scheme.Codecs.EncoderForVersion(encoder, v1.SchemeGroupVersion)
+	encoder = scheme.Codecs.EncoderForVersion(encoder, gv)
```

Comment:
Here, I'd check if the `runtime.Object` argument has a non-empty GroupVersion. If so, I'd use this `EncoderForVersion` func to get the encoder. If not, I'd do something similar to my code sample (use `scheme.ObjectKinds(obj)`) to try to find the `GroupVersionKind` for the object (this would apply if it's a strongly-typed object, that doesn't have `TypeMeta` filled in).  Have to decide how to handle an `error` or <1 GVK being returned from that function -- probably just return the generic `serializerInfo.PrettySerializer` or `serializerInfo.Serializer`.

Reply:
I tried out returning the generic encoder when no GVK was found by removing the velero scheme and running `velero backup create` - it still worked, even without the type meta attached.

I don't believe `scheme.ObjectKinds` can return a slice of len(0), due to the fact that all the versions have to be explicitly registered in a map, and the `Kind` key missing from that map generates a not found error.